src/solace_agent_mesh/gateway/http_sse/services/audio_service.py

```diff
@@ -0,0 +1,1154 @@
+                done.set()
+            
+            def stopped_handler(evt):
+                done.set()
+            
+            # Connect callbacks
+            recognizer.recognized.connect(recognized_handler)
+            recognizer.canceled.connect(canceled_handler)
+            recognizer.session_stopped.connect(stopped_handler)
+            
+            # Start continuous recognition
+            await asyncio.to_thread(recognizer.start_continuous_recognition)
+            
+            # Wait for recognition to complete
+            await done.wait()
+            
+            # Stop recognition
+            await asyncio.to_thread(recognizer.stop_continuous_recognition)
+            
+            # Combine all recognized text
+            if not transcribed_texts:
+                log.warning("[AudioService] No speech could be recognized")
+                raise HTTPException(400, "No speech detected in audio. Please try speaking louder or longer.")
+            
+            full_transcription = " ".join(transcribed_texts).strip()
+            
+            return TranscriptionResult(
+                text=full_transcription,
+                language=language,
+                duration=0.0  # Duration not available in continuous mode
+            )
+                
+        except HTTPException:
+            raise
+        except Exception as e:
+            log.exception("[AudioService] Azure STT error: %s", e)
+            raise HTTPException(500, f"Azure STT failed: {str(e)}")
+        finally:
+            # Clean up WAV temp file if created
+            if wav_temp_path:
+                try:
+                    import os
+                    os.unlink(wav_temp_path)
+                except Exception as e:
+                    log.warning("[AudioService] Failed to delete WAV temp file: %s", e)
+    
+    async def transcribe_audio(
+        self,
+        audio_file: UploadFile,
+        user_id: str,
+        session_id: str,
+        app_name: str = "webui",
+        provider: Optional[str] = None
+    ) -> TranscriptionResult:
+        """
+        Transcribe audio file to text using configured STT service.
+        Routes to appropriate provider (OpenAI, Azure, etc.).
+        
+        Args:
+            audio_file: Uploaded audio file
+            user_id: User identifier
+            session_id: Session identifier
+            app_name: Application name
+            provider: Optional provider override (openai, azure)
+            
+        Returns:
+            TranscriptionResult with transcribed text
+            
+        Raises:
+            HTTPException: If transcription fails
+        """
+        
+        try:
+            # Validate file
+            if not audio_file.filename:
+                raise HTTPException(400, "No filename provided")
+            
+            # Check file size (max 25MB)
+            content = await audio_file.read()
```

Comment:
I think we should check the file size before reading it in completely or else could cause oom potentially if the file is massive?